File: in_Java/1347 Minimum Number of Steps to Make Two Strings Anagram.java
class Solution {
    public int minSteps(String s, String t) {
        Map<Character,Integer> hm1 = new HashMap<>();
        Map<Character,Integer> hm2 = new HashMap<>();
        for(char c : s.toCharArray()){
            if(!hm1.containsKey(c)) hm1.put(c,1);
            else hm1.put(c, hm1.get(c)+1);
        }
        for(char c : t.toCharArray()){
            if(!hm2.containsKey(c)) hm2.put(c,1);
            else hm2.put(c, hm2.get(c)+1);
        }
        int res = 0;
        for(char k : hm1.keySet()){
            if(!hm2.containsKey(k)) res += hm1.get(k);
        }
        for(char k : hm2.keySet()){
            if(hm1.containsKey(k) && hm1.get(k) > hm2.get(k)) res += hm1.get(k) - hm2.get(k);
        }
        return res;
    }
}

File: in_Python/1347 Minimum Number of Steps to Make Two Strings Anagram.py
class Solution(object):
    def minSteps(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: int
        """
        res = 0
        d1 = {}
        d2 = {}
        for i in range(len(s)):
            c1 = s[i]
            c2 = t[i]
            if c1 not in d1:
                d1[c1] = 1
            else:
                d1[c1] += 1
            if c2 not in d2:
                d2[c2] = 1
            else:
                d2[c2] += 1
        for c in d2:
            if c in d1 and d2[c] > d1[c]:
                res += d2[c] - d1[c]
            if c not in d1:
                res += d2[c]
        return res